### PR TITLE
New API for Retrieval / Bulk Ingest

### PR DESCRIPTION
**Pull request by heckdevice**
Implemented the following two apis:

1. /db/{tablename} GET operation internally transforms to a Select * from {tablename} and returns the JSON

2. /db/bulk POST operation bulk ingests the data into tables. The payload has to be in the following format:
Taking e.g. of table foo with columns: id and name
{
"data":
[
 {
  "table":"foo",
  "payload":
  [{"name":"testdata6","id":"601"},
  {"name":"testdata7","id":"701"}]
 }
]
}

The same can be used to insert into multiple tables as well.

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks for the PR.

File: main.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ import (
 	"runtime/pprof"
 	"strings"
 	"time"
-
+	
 	"github.com/otoolep/rqlite/log"
-	"github.com/otoolep/rqlite/server"
+	"github.com/jsrCorp/rqlite/server"
```
Comment (otoolep) [inline main.go]: You need to revert this change.

File: server/server.go [hunk 2]
```diff
@@ -313,7 +314,9 @@ func (s *Server) ListenAndServe(leader string) error {
 	s.router.HandleFunc("/diagnostics", s.serveDiagnostics).Methods("GET")
 	s.router.HandleFunc("/raft", s.serveRaftInfo).Methods("GET")
 	s.router.HandleFunc("/db", s.readHandler).Methods("GET")
+	s.router.HandleFunc("/db/{tablename}", s.tableReadHandler).Methods("GET")
 	s.router.HandleFunc("/db", s.writeHandler).Methods("POST")
+	s.router.HandleFunc("/db/ingest", s.tableWriteHandler).Methods("POST")
```
Comment (otoolep) [inline server/server.go]: How about using `/db/bulk` instead? Both `POST` endpoints allow data to be "ingested".

File: server/server.go [hunk 3]
```diff
@@ -389,7 +392,7 @@ func (s *Server) joinHandler(w http.ResponseWriter, req *http.Request) {
 }
 
 func (s *Server) readHandler(w http.ResponseWriter, req *http.Request) {
-	log.Tracef("readHandler for URL: %s", req.URL)
+	log.Infof("readHandler for URL: %s", req.URL)
```
Comment (otoolep) [inline server/server.go]: Did you mean to change this?

File: server/server.go [hunk 4]
```diff
@@ -436,7 +439,48 @@ func (s *Server) readHandler(w http.ResponseWriter, req *http.Request) {
...
 	}
 }
+func (s *Server) tableReadHandler(w http.ResponseWriter, req *http.Request) {
+	log.Infof("tableReadHandler for URL: %s", req.URL)
+	s.metrics.queryReceived.Inc(1)
+
+	var failures = make([]FailedSqlStmt, 0)
+	vars := mux.Vars(req)
+	tablename := vars["tablename"]
+	log.Infof("Getting data from table %s", tablename)
```
Comment (otoolep) [inline server/server.go]: I think this should be trace.

File: server/server.go [hunk 5]
```diff
@@ -436,7 +439,48 @@ func (s *Server) readHandler(w http.ResponseWriter, req *http.Request) {
...
 
+	pretty, _ := isPretty(req)
+	var b []byte
+	if pretty {
+		b, err = json.MarshalIndent(rr, "", "    ")
+	} else {
+		b, err = json.Marshal(rr)
+	}
+	if err != nil {
+		log.Tracef("Failed to marshal JSON data: %s", err.Error())
```
Comment (otoolep) [inline server/server.go]: Why not log this at error lvel?

File: server/server.go [hunk 6]
```diff
@@ -436,7 +439,48 @@ func (s *Server) readHandler(w http.ResponseWriter, req *http.Request) {
...
+		http.Error(w, err.Error(), http.StatusBadRequest) // Internal error actually
```
Comment (otoolep) [inline server/server.go]: It's more Go-style to return from this clause, and have no `else` clause. Can you make that change? (And in other places where it matter)?

**Comment by otoolep** [comment]
I have feedback, but am up for merging this once some changes are made.

Once we're both happy, I will ask that you squash the commits into a single commit.

File: server/server.go (hunk 2 above)
Comment (heckdevice) [inline server/server.go]: Thats a valid point. OK

File: server/server.go (hunk 3 above)
Comment (heckdevice) [inline server/server.go]: Mistake. Will revert.

File: server/server.go (hunk 5 above)
Comment (heckdevice) [inline server/server.go]: Korrect. Will do

File: server/server.go (hunk 6 above)
Comment (heckdevice) [inline server/server.go]: Ok. my first go exp so... :)
